nautobot/ipam/utils/migrations.py

```diff
@@ -731,3 +731,27 @@ def ensure_correct_prefix_broadcast(apps):
                 )
             prefix.broadcast = true_broadcast
             prefix.save()
+
+
+def increment_names_of_records_with_similar_names(model: models.Model):
+    """
+    This function increments the names of records with similar names in a given model.
+    """
+    cache = set()
+    records_to_update = []
+    for instance in model.objects.all():
```

Comment:
with the potential for a large number of records, should we be using `iterator` here? Compare `nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py`